add keep parameter

zkamvar opened the pull request:
I am in a situation where I want to archive a certain subset of repositories before I make irreversible changes to them.

This keep parameter will help me avoid downloading hundreds of repositories that I do not wish to archive.

zkamvar committed:
add keep parameter

maelle commented on NEWS.md:
```diff
@@ -1,5 +1,7 @@
 # gitcellar 0.0.0.9000
 
+* Parameter `keep` allows users to define which repositories should be archived
```
Please add your username here :smile_cat:

zkamvar replied:
```suggestion
* Parameter `keep` allows users to define which repositories should be archived (#13, @zkamvar)
```

maelle reviewed (changes requested):
Thanks a ton! Two small comments.

zkamvar committed:
Update NEWS.md

zkamvar commented:
Done! I think? I only saw one comment

maelle commented:
Oops, my other comment was, what should happen is some of the "keep" names are NOT in the repos? Should there be a warning?

zkamvar commented:
> Oops, my other comment was, what should happen is some of the "keep" names are NOT in the repos? Should there be a warning?

Oh! Good point! I'll add that logic in and throw an error because the keep parameter is opt-in, so the user will likely want to know immediately if they are requesting repositories that are not passed through for a process that could take a long time.

zkamvar committed:
add error for extra repos in keep

Since keep is an optional parameter, then we need to be mindful of the
fact that users who use this option intend for these specific
repositories to be archived. If there is a repository that does not
exist, it is better to stop early and alert the user instead of issuing
a warning and dropping the non-existent repository, which would eat up
bandwidth when there might be a larger error that would need to be
addressed.

zkamvar commented:
This is ready for another round of review when you are back :)

maelle committed:
Merge branch 'main' into add-filter-option

maelle commented:
Thank you!!
